packages/engine/src/avatar/AvatarSystem.ts

```diff
@@ -75,6 +77,16 @@ function avatarActionReceptor(action) {
           inputData.controllerGripRightParent.add(inputData.controllerGripRight)
 
           addComponent(entity, XRInputSourceComponent, inputData as any)
+
+          dispatchAction(
+            Engine.instance.store,
+            AvatarInputSettingsAction.setControlType(a.avatarInputControllerType as any)
+          )
+          // This is required because state is updated in the next frame and some functionality that requires updated controller type might run in the current frame.
+          // This component can be used at such places.
+          addComponent(entity, AvatarInputControllerTypeUpdatePendingComponent, {
```

Comment:
let's try and refactor around not needing this component, we should rely more heavily on actions than adding and removing components

Reply:
"some functionality that requires updated controller type might run in the current frame" which functionality?
"This component can be used at such places." What places?

Reply:
> "some functionality that requires updated controller type might run in the current frame" which functionality? "This component can be used at such places." What places?

Hey sorry for the confusion I have updated the comment now to be more understandable.